resources/dw/virtualwarehouse/impala/resource_impala_vw_test.go

```diff
@@ -0,0 +1,343 @@
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/client/operations"
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/models"
+	mocks "github.com/cloudera/terraform-provider-cdp/mocks/github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/client/operations"
+)
+
+var testImpalaSchema = schema.Schema{
+	Attributes: map[string]schema.Attribute{
+		"id": schema.StringAttribute{
+			Computed: true,
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.UseStateForUnknown(),
+			},
+		},
+		"cluster_id": schema.StringAttribute{
+			Required:            true,
+			MarkdownDescription: "The id of the CDW Cluster which the Impala Virtual Warehouse is attached to.",
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.UseStateForUnknown(),
+			},
+		},
+		"database_catalog_id": schema.StringAttribute{
+			Required:            true,
+			MarkdownDescription: "The id of the Database Catalog which the Impala Virtual Warehouse is attached to.",
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.UseStateForUnknown(),
+			},
+		},
+		"name": schema.StringAttribute{
+			Required:            true,
+			MarkdownDescription: "The name of the Impala Virtual Warehouse.",
+		},
+		"last_updated": schema.StringAttribute{
+			Computed:            true,
+			MarkdownDescription: "Timestamp of the last Terraform update of the order.",
+		},
+		"status": schema.StringAttribute{
+			Computed:            true,
+			MarkdownDescription: "The status of the database catalog.",
+		},
+		"polling_options": schema.SingleNestedAttribute{
+			MarkdownDescription: "Polling related configuration options that could specify various values that will be used during CDP resource creation.",
+			Optional:            true,
+			Attributes: map[string]schema.Attribute{
+				"async": schema.BoolAttribute{
+					MarkdownDescription: "Boolean value that specifies if Terraform should wait for resource creation/deletion.",
+					Optional:            true,
+					Computed:            true,
+					Default:             booldefault.StaticBool(false),
+					PlanModifiers: []planmodifier.Bool{
+						boolplanmodifier.UseStateForUnknown(),
+					},
+				},
+				"polling_timeout": schema.Int64Attribute{
+					MarkdownDescription: "Timeout value in minutes that specifies for how long should the polling go for resource creation/deletion.",
+					Default:             int64default.StaticInt64(40),
+					Computed:            true,
+					Optional:            true,
+				},
+				"call_failure_threshold": schema.Int64Attribute{
+					MarkdownDescription: "Threshold value that specifies how many times should a single call failure happen before giving up the polling.",
+					Default:             int64default.StaticInt64(3),
+					Computed:            true,
+					Optional:            true,
+				},
+			},
+		},
+	},
+}
+
+type MockTransport struct {
+	runtime.ClientTransport
+}
+
+func NewDwApi(client *mocks.MockDwClientService) *impalaResource {
```

Comment:
exported function with the unexported return type; use newDwApi to unexport the function

Reply:
sure, I would unexport the function